### ISSUE TITLE
Working with Minisforum UM890Pro 780M graphics

### ISSUE THREAD

**Issue by drvetcosta** [isc30/ryzen-7000-series-proxmox#61]
Followed your instructions 2 or 3 times, and it is working. 
i am joining the vbios file for the Ryzen 9 8945HS/Radeon 780M
[vbios_8945.bin.patch](https://github.com/user-attachments/files/16928837/vbios_8945.bin.patch)

**Comment by drvetcosta** [comment]
Next step : win 11
and exctraction of GOP driver from the bios file

**Comment by BLNJ** [comment]
@drvetcosta Did you have any success with Windows 11? Thank you for the vbios patch aswell

**Comment by drvetcosta** [comment]
succes with windows 11 yes, and since i have installed windwos 11 without windows 10 before, and all good.

**Comment by dextouu** [comment]
hello, did you manage to get the rom for the audio, why on the manufacturer's website, no bios is available. if you have it, would it be possible to provide it?

**Comment by drvetcosta** [comment]
I asked minisforum support for the files, but i don't know exactly how to do it, what i exctract from the bios isn't an EFI file ...

**Comment by dextouu** [comment]
[Hello, I also asked by email with support for the BIOS. They sent me the email. I managed to extract thanks to the ubu software. The only problem for me, when I activate the driver audio. Promox Crash (probably a panic kernel). Can you test on your side, I provide you with the file. To see if it crashes, at home  too.

[[https://github.com/dextouu/Proxmox-igpu-passthrough](url)

**Comment by drvetcosta** [comment]
@dextouu I will try right now: did you add the 2 other pci devices (audio) in the VM ?

**Comment by dextouu** [comment]
Yes

**Comment by dextouu** [comment]
![Screenshot_20240924-203649](https://github.com/user-attachments/assets/f62951c1-faa9-444f-a669-e646c1bdb4b6)

**Comment by drvetcosta** [comment]
working on my VM.

there are 2 pci devices linked to the audio system to add. maybe 0000:C5:00.5 and 6

**Comment by dextouu** [comment]
By adding these two more, it works. But you only give the audio rom to the   0000:C5:00.1. correct?

**Comment by dextouu** [comment]
@drvetcosta: Thank you, it works with the addition of the two PCI devices.😀

**Comment by isc30** [comment]
Hey, thats great news, can you please open a Pull Request with that note
for others reading the wiki pls?

**Comment by dextouu** [comment]
Yes, I will do it tomorrow, I'm going to bed for work. I will do it tomorrow afternoon.

**Comment by AutomationD** [comment]
Thanks for submitting the patch!

I have Beelink SER8 with Ryzen 7 / 780M 8945HS and had spent 3 days with trying to get the iGPU passthrough going. Tried so many combos and quite lost. 

After adding all configs and booting the mini-pc I see a bit of dmesg but then it freezes (I think video gets disconnected from the amdgpu driver and switches to vfio-pci. 

I've switched to Ubuntu, just in case. My VM is configured like that (thousands of different iterations...):
```
root@pve:~# cat /etc/pve/qemu-server/100.conf
agent: 1
balloon: 0
bios: ovmf
boot: order=scsi0;ide2;net0
cores: 8
cpu: host
efidisk0: local-lvm:vm-100-disk-0,efitype=4m,pre-enrolled-keys=1,size=4M
hookscript: local:snippets/resetpcie.sh
hostpci0: 0000:66:00.0,pcie=1,rombar=0,romfile=vbios_8745hs.bin
hostpci1: 0000:66:00.1,pcie=1,rombar=0
hostpci2: 0000:66:00.5,pcie=1,rombar=0
hostpci3: 0000:66:00.6,pcie=1,rombar=0
ide2: none,media=cdrom
machine: q35
memory: 16384
meta: creation-qemu=9.0.2,ctime=1739556374
name: play
net0: virtio=BC:24:11:42:CA:1D,bridge=vmbr0,firewall=1
numa: 0
ostype: l26
scsi0: vm:vm-100-disk-3,cache=writeback,iothread=1,size=32G,ssd=1
scsihw: virtio-scsi-single
smbios1: uuid=862bd4eb-4d2b-4c4f-92a9-231af5d7cbcd
sockets: 1
vga: none
vmgenid: c945690f-4e86-43de-840e-d1edff2af813
````


To troubleshoot , I'm testing it with direct bind/unbind
```
echo 0000:66:00.0 > /sys/bus/pci/drivers/vfio-pci/unbind
modprobe amdgpu
echo 0000:66:00.0 > /sys/bus/pci/drivers/amdgpu/bind
```
It switches the screen to show proxmox login prompt (which tells me I am able to tell amdgpu driver to take the gpu?)

Next I try to bind it back to vfio-pci
```
echo 0000:66:00.0 > /sys/bus/pci/drivers/amdgpu/unbind # Disconnects ssh session from where this is executed too! and shows stacktrace on the dmesg..

<details><summary>Details</summary>
<p>
Feb 14 21:32:38 pve kernel: vfio-pci 0000:66:00.0: vgaarb: VGA decodes changed: olddecodes=io+mem,decodes=io+mem:owns=none
Feb 14 21:32:44 pve kernel: [drm] amdgpu kernel modesetting enabled.
Feb 14 21:32:44 pve kernel: amdgpu: Virtual CRAT table created for CPU
Feb 14 21:32:44 pve kernel: amdgpu: Topology: Add CPU node
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: enabling device (0006 -> 0007)
Feb 14 21:32:44 pve kernel: [drm] initializing kernel modesetting (IP DISCOVERY 0x1002:0x1900 0x1F66:0x0031 0xB3).
Feb 14 21:32:44 pve kernel: [drm] register mmio base: 0xDC500000
Feb 14 21:32:44 pve kernel: [drm] register mmio size: 524288
Feb 14 21:32:44 pve kernel: [drm] add ip block number 0 <soc21_common>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 1 <gmc_v11_0>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 2 <ih_v6_0>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 3 <psp>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 4 <smu>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 5 <dm>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 6 <gfx_v11_0>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 7 <sdma_v6_0>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 8 <vcn_v4_0>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 9 <jpeg_v4_0>
Feb 14 21:32:44 pve kernel: [drm] add ip block number 10 <mes_v11_0>
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: Fetched VBIOS from VFCT
Feb 14 21:32:44 pve kernel: amdgpu: ATOM BIOS: 113-PHXGENERIC-001
Feb 14 21:32:44 pve kernel: [drm] VCN(0) encode/decode are enabled in VM mode
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: [drm:jpeg_v4_0_early_init [amdgpu]] JPEG decode is enabled in VM mode
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: vgaarb: deactivate vga console
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: Trusted Memory Zone (TMZ) feature enabled
Feb 14 21:32:44 pve kernel: [drm] vm size is 262144 GB, 4 levels, block size is 9-bit, fragment size is 9-bit
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: VRAM: 4096M 0x0000008000000000 - 0x00000080FFFFFFFF (4096M used)
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: GART: 512M 0x00007FFF00000000 - 0x00007FFF1FFFFFFF
Feb 14 21:32:44 pve kernel: [drm] Detected VRAM RAM=4096M, BAR=4096M
Feb 14 21:32:44 pve kernel: [drm] RAM width 128bits DDR5
Feb 14 21:32:44 pve kernel: [drm] amdgpu: 4096M of VRAM memory ready
Feb 14 21:32:44 pve kernel: [drm] amdgpu: 46153M of GTT memory ready.
Feb 14 21:32:44 pve kernel: [drm] GART: num cpu pages 131072, num gpu pages 131072
Feb 14 21:32:44 pve kernel: [drm] PCIE GART of 512M enabled (table at 0x00000080FFD00000).
Feb 14 21:32:44 pve kernel: [drm] Loading DMUB firmware via PSP: version=0x08004300
Feb 14 21:32:44 pve kernel: [drm] Found VCN firmware Version ENC: 1.22 DEC: 9 VEP: 0 Revision: 3
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: Will use PSP to load VCN firmware
Feb 14 21:32:44 pve kernel: [drm] reserve 0x4000000 from 0x80f8000000 for PSP TMR
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: RAS: optional ras ta ucode is not available
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: RAP: optional rap ta ucode is not available
Feb 14 21:32:44 pve kernel: amdgpu 0000:66:00.0: amdgpu: SECUREDISPLAY: securedisplay ta ucode is not available
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: SMU is initialized successfully!
Feb 14 21:32:45 pve kernel: [drm] Seamless boot condition check passed
Feb 14 21:32:45 pve kernel: [drm] Display Core v3.2.266 initialized on DCN 3.1.4
Feb 14 21:32:45 pve kernel: [drm] DP-HDMI FRL PCON supported
Feb 14 21:32:45 pve kernel: [drm] DMUB hardware initialized: version=0x08004300
Feb 14 21:32:45 pve kernel: [drm] kiq ring mec 3 pipe 1 q 0
Feb 14 21:32:45 pve kernel: [drm] VCN decode and encode initialized successfully(under DPG Mode).
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: [drm:jpeg_v4_0_hw_init [amdgpu]] JPEG decode initialized successfully.
Feb 14 21:32:45 pve kernel: amdgpu: HMM registered 4096MB device memory
Feb 14 21:32:45 pve kernel: kfd kfd: amdgpu: Allocated 3969056 bytes on gart
Feb 14 21:32:45 pve kernel: kfd kfd: amdgpu: Total number of KFD nodes to be created: 1
Feb 14 21:32:45 pve kernel: amdgpu: Virtual CRAT table created for GPU
Feb 14 21:32:45 pve kernel: amdgpu: Topology: Add dGPU node [0x1900:0x1002]
Feb 14 21:32:45 pve kernel: kfd kfd: amdgpu: added device 1002:1900
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: SE 1, SH per SE 2, CU per SH 6, active_cu_number 12
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring gfx_0.0.0 uses VM inv eng 0 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 6 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 7 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 8 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 9 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 10 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 11 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring sdma0 uses VM inv eng 12 on hub 0
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring vcn_unified_0 uses VM inv eng 0 on hub 8
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring jpeg_dec uses VM inv eng 1 on hub 8
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: amdgpu: ring mes_kiq_3.1.0 uses VM inv eng 13 on hub 0
Feb 14 21:32:45 pve kernel: [drm] ring gfx_32768.1.1 was added
Feb 14 21:32:45 pve kernel: [drm] ring compute_32768.2.2 was added
Feb 14 21:32:45 pve kernel: [drm] ring sdma_32768.3.3 was added
Feb 14 21:32:45 pve kernel: [drm] ring gfx_32768.1.1 ib test pass
Feb 14 21:32:45 pve kernel: [drm] ring compute_32768.2.2 ib test pass
Feb 14 21:32:45 pve kernel: [drm] ring sdma_32768.3.3 ib test pass
Feb 14 21:32:45 pve kernel: [drm] Initialized amdgpu 3.57.0 20150101 for 0000:66:00.0 on minor 0
Feb 14 21:32:45 pve kernel: fbcon: amdgpudrmfb (fb0) is primary device
Feb 14 21:32:45 pve kernel: [drm] DSC precompute is not needed.
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: [drm] REG_WAIT timeout 1us * 100000 tries - optc314_disable_crtc line:146
Feb 14 21:32:45 pve kernel: Console: switching to colour frame buffer device 240x67
Feb 14 21:32:45 pve kernel: amdgpu 0000:66:00.0: [drm] fb0: amdgpudrmfb frame buffer device
Feb 14 21:33:59 pve kernel: Console: switching to colour dummy device 80x25
Feb 14 21:33:59 pve kernel: amdgpu 0000:66:00.0: [drm] REG_WAIT timeout 1us * 100000 tries - optc1_wait_for_state line:839
Feb 14 21:34:00 pve kernel: amdgpu 0000:66:00.0: [drm] REG_WAIT timeout 1us * 100000 tries - optc1_wait_for_state line:839
Feb 14 21:34:00 pve kernel: amdgpu 0000:66:00.0: [drm] REG_WAIT timeout 1us * 100000 tries - optc1_wait_for_state line:839
Feb 14 21:34:00 pve kernel: amdgpu 0000:66:00.0: amdgpu: amdgpu: finishing device.
Feb 14 21:34:00 pve kernel: ------------[ cut here ]------------
Feb 14 21:34:00 pve kernel: kernel BUG at mm/slub.c:553!
Feb 14 21:34:00 pve kernel: invalid opcode: 0000 [#1] PREEMPT SMP NOPTI
Feb 14 21:34:00 pve kernel: CPU: 5 PID: 1272 Comm: bash Tainted: P           O       6.8.4-2-pve #1
Feb 14 21:34:00 pve kernel: Hardware name: AZW SER8/SER8, BIOS HPT.8xxx.SER8.V029.P8C0M0C15.08.Link 10/23/2024
Feb 14 21:34:00 pve kernel: RIP: 0010:kfree+0x241/0x2e0
Feb 14 21:34:00 pve kernel: Code: ff ff 4c 8d 62 ff e9 5f fe ff ff 4d 89 f1 41 b8 01 00 00 00 48 89 d9 48 89 da 4c 89 e6 4c 89 ef e8 a4 fa ff ff e9 df fe ff ff <0f> 0b 49 83 e0 fc b9 01 00 00 00 48 89 e2 4c 89 e6 4c 89 ef e8 16
Feb 14 21:34:00 pve kernel: RSP: 0018:ffffade5c15e3af0 EFLAGS: 00010246
Feb 14 21:34:00 pve kernel: RAX: ffffa09d45c1b200 RBX: ffffa09d45c1b200 RCX: ffffa09d45c1b280
Feb 14 21:34:00 pve kernel: RDX: 0000000000014005 RSI: ffffffffc2473a55 RDI: ffffa09d45c1b200
Feb 14 21:34:00 pve kernel: RBP: ffffade5c15e3b28 R08: 0000000000000000 R09: 0000000000000000
Feb 14 21:34:00 pve kernel: R10: 0000000000000000 R11: 0000000000000000 R12: ffffdb38c4170680
Feb 14 21:34:00 pve kernel: R13: ffffa09d40045d00 R14: ffffffffc2473a55 R15: ffffa09da583c818
Feb 14 21:34:00 pve kernel: FS:  000074ae1914e740(0000) GS:ffffa0b341c80000(0000) knlGS:0000000000000000
Feb 14 21:34:00 pve kernel: CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Feb 14 21:34:00 pve kernel: CR2: 00005a9037253508 CR3: 00000001089ca000 CR4: 0000000000f50ef0
Feb 14 21:34:00 pve kernel: PKRU: 55555554
Feb 14 21:34:00 pve kernel: Call Trace:
Feb 14 21:34:00 pve kernel:  <TASK>
Feb 14 21:34:00 pve kernel:  ? show_regs+0x6d/0x80
Feb 14 21:34:00 pve kernel:  ? die+0x37/0xa0
Feb 14 21:34:00 pve kernel:  ? do_trap+0xd4/0xf0
Feb 14 21:34:00 pve kernel:  ? do_error_trap+0x71/0xb0
Feb 14 21:34:00 pve kernel:  ? kfree+0x241/0x2e0
Feb 14 21:34:00 pve kernel:  ? exc_invalid_op+0x52/0x80
Feb 14 21:34:00 pve kernel:  ? kfree+0x241/0x2e0
Feb 14 21:34:00 pve kernel:  ? asm_exc_invalid_op+0x1b/0x20
Feb 14 21:34:00 pve kernel:  ? dcn10_link_encoder_destroy+0x15/0x30 [amdgpu]
Feb 14 21:34:00 pve kernel:  ? dcn10_link_encoder_destroy+0x15/0x30 [amdgpu]
Feb 14 21:34:00 pve kernel:  ? kfree+0x241/0x2e0
Feb 14 21:34:00 pve kernel:  dcn10_link_encoder_destroy+0x15/0x30 [amdgpu]
Feb 14 21:34:00 pve kernel:  link_destroy+0xdd/0x140 [amdgpu]
Feb 14 21:34:00 pve kernel:  dc_destruct+0x97/0x290 [amdgpu]
Feb 14 21:34:00 pve kernel:  dc_destroy+0x15/0x30 [amdgpu]
Feb 14 21:34:00 pve kernel:  amdgpu_dm_fini+0x1b8/0x240 [amdgpu]
Feb 14 21:34:00 pve kernel:  dm_hw_fini+0x22/0x30 [amdgpu]
Feb 14 21:34:00 pve kernel:  amdgpu_device_fini_hw+0x2ba/0x460 [amdgpu]
Feb 14 21:34:00 pve kernel:  amdgpu_driver_unload_kms+0x4b/0x70 [amdgpu]
Feb 14 21:34:00 pve kernel:  amdgpu_pci_remove+0x3c/0x80 [amdgpu]
Feb 14 21:34:00 pve kernel:  pci_device_remove+0x3e/0xb0
Feb 14 21:34:00 pve kernel:  device_remove+0x40/0x80
Feb 14 21:34:00 pve kernel:  device_release_driver_internal+0x20b/0x270
Feb 14 21:34:00 pve kernel:  ? bus_find_device+0xb8/0xf0
Feb 14 21:34:00 pve kernel:  device_driver_detach+0x14/0x20
Feb 14 21:34:00 pve kernel:  unbind_store+0xac/0xc0
Feb 14 21:34:00 pve kernel:  drv_attr_store+0x21/0x50
Feb 14 21:34:00 pve kernel:  sysfs_kf_write+0x3b/0x60
Feb 14 21:34:00 pve kernel:  kernfs_fop_write_iter+0x130/0x210
Feb 14 21:34:00 pve kernel:  vfs_write+0x2a5/0x480
Feb 14 21:34:00 pve kernel:  ksys_write+0x73/0x100
Feb 14 21:34:00 pve kernel:  __x64_sys_write+0x19/0x30
Feb 14 21:34:00 pve kernel:  do_syscall_64+0x84/0x180
Feb 14 21:34:00 pve kernel:  ? do_syscall_64+0x93/0x180
Feb 14 21:34:00 pve kernel:  ? srso_alias_return_thunk+0x5/0xfbef5
Feb 14 21:34:00 pve kernel:  ? syscall_exit_to_user_mode+0x86/0x260
Feb 14 21:34:00 pve kernel:  ? srso_alias_return_thunk+0x5/0xfbef5
Feb 14 21:34:00 pve kernel:  ? do_syscall_64+0x93/0x180
Feb 14 21:34:00 pve kernel:  ? do_syscall_64+0x93/0x180
Feb 14 21:34:00 pve kernel:  ? srso_alias_return_thunk+0x5/0xfbef5
Feb 14 21:34:00 pve kernel:  ? irqentry_exit+0x43/0x50
Feb 14 21:34:00 pve kernel:  ? srso_alias_return_thunk+0x5/0xfbef5
Feb 14 21:34:00 pve kernel:  ? exc_page_fault+0x94/0x1b0
Feb 14 21:34:00 pve kernel:  entry_SYSCALL_64_after_hwframe+0x73/0x7b
Feb 14 21:34:00 pve kernel: RIP: 0033:0x74ae192492c0
Feb 14 21:34:00 pve kernel: Code: 40 00 48 8b 15 41 9b 0d 00 f7 d8 64 89 02 48 c7 c0 ff ff ff ff eb b7 0f 1f 00 80 3d 21 23 0e 00 00 74 17 b8 01 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 58 c3 0f 1f 80 00 00 00 00 48 83 ec 28 48 89
Feb 14 21:34:00 pve kernel: RSP: 002b:00007fff3b142f18 EFLAGS: 00000202 ORIG_RAX: 0000000000000001
Feb 14 21:34:00 pve kernel: RAX: ffffffffffffffda RBX: 000000000000000d RCX: 000074ae192492c0
Feb 14 21:34:00 pve kernel: RDX: 000000000000000d RSI: 00005a903725fdd0 RDI: 0000000000000001
Feb 14 21:34:00 pve kernel: RBP: 00005a903725fdd0 R08: 0000000000000007 R09: 0000000000000073
Feb 14 21:34:00 pve kernel: R10: 0000000000000000 R11: 0000000000000202 R12: 000000000000000d
Feb 14 21:34:00 pve kernel: R13: 000074ae19324760 R14: 000000000000000d R15: 000074ae1931f9e0
Feb 14 21:34:00 pve kernel:  </TASK>
Feb 14 21:34:00 pve kernel: Modules linked in: amdgpu amdxcp drm_exec gpu_sched drm_buddy drm_suballoc_helper drm_ttm_helper ttm drm_display_helper cec rc_core i2c_algo_bit ebtable_filter ebtables ip_set ip6table_raw iptable_raw ip6table_filter ip6_tables iptable_filter nf_tables bonding tls softdog sunrpc nfnetlink_log nfnetlink binfmt_misc intel_rapl_msr intel_rapl_common snd_soc_dmic snd_soc_ps_mach snd_ps_pdm_dma snd_sof_amd_acp63 snd_sof_amd_vangogh snd_sof_amd_rembrandt edac_mce_amd snd_sof_amd_renoir snd_sof_amd_acp snd_sof_pci snd_sof_xtensa_dsp iwlmvm snd_sof kvm_amd snd_hda_codec_realtek snd_hda_codec_generic mac80211 snd_sof_utils kvm snd_soc_core libarc4 snd_hda_intel snd_compress ac97_bus snd_intel_dspcfg snd_pcm_dmaengine snd_intel_sdw_acpi btusb crct10dif_pclmul snd_pci_ps polyval_clmulni btrtl snd_hda_codec polyval_generic snd_rpl_pci_acp6x btintel ghash_clmulni_intel snd_acp_pci sha256_ssse3 btbcm snd_hda_core sha1_ssse3 snd_acp_legacy_common btmtk snd_pci_acp6x snd_hwdep aesni_intel bluetooth iwlwifi snd_pcm
Feb 14 21:34:00 pve kernel:  snd_pci_acp5x crypto_simd snd_rn_pci_acp3x cryptd snd_timer joydev snd_acp_config snd snd_soc_acpi ecdh_generic input_leds cfg80211 rapl pcspkr k10temp wmi_bmof soundcore snd_pci_acp3x ccp ecc amd_pmc mac_hid zfs(PO) spl(O) vhost_net vhost vhost_iotlb tap vfio_pci vfio_pci_core irqbypass vfio_iommu_type1 vfio iommufd efi_pstore dmi_sysfs ip_tables x_tables autofs4 hid_logitech_hidpp hid_logitech_dj hid_generic usbmouse usbkbd usbhid dm_thin_pool dm_persistent_data dm_bio_prison dm_bufio uas usb_storage btrfs blake2b_generic xor raid6_pq libcrc32c xhci_pci nvme xhci_pci_renesas crc32_pclmul thunderbolt xhci_hcd nvme_core r8169 i2c_piix4 realtek nvme_auth video wmi i2c_hid_acpi i2c_hid hid
Feb 14 21:34:00 pve kernel: ---[ end trace 0000000000000000 ]---
Feb 14 21:34:00 pve kernel: RIP: 0010:kfree+0x241/0x2e0
Feb 14 21:34:00 pve kernel: Code: ff ff 4c 8d 62 ff e9 5f fe ff ff 4d 89 f1 41 b8 01 00 00 00 48 89 d9 48 89 da 4c 89 e6 4c 89 ef e8 a4 fa ff ff e9 df fe ff ff <0f> 0b 49 83 e0 fc b9 01 00 00 00 48 89 e2 4c 89 e6 4c 89 ef e8 16
Feb 14 21:34:00 pve kernel: RSP: 0018:ffffade5c15e3af0 EFLAGS: 00010246
Feb 14 21:34:00 pve kernel: RAX: ffffa09d45c1b200 RBX: ffffa09d45c1b200 RCX: ffffa09d45c1b280
Feb 14 21:34:00 pve kernel: RDX: 0000000000014005 RSI: ffffffffc2473a55 RDI: ffffa09d45c1b200
Feb 14 21:34:00 pve kernel: RBP: ffffade5c15e3b28 R08: 0000000000000000 R09: 0000000000000000
Feb 14 21:34:00 pve kernel: R10: 0000000000000000 R11: 0000000000000000 R12: ffffdb38c4170680
Feb 14 21:34:00 pve kernel: R13: ffffa09d40045d00 R14: ffffffffc2473a55 R15: ffffa09da583c818
Feb 14 21:34:00 pve kernel: FS:  000074ae1914e740(0000) GS:ffffa0b341c80000(0000) knlGS:0000000000000000
Feb 14 21:34:00 pve kernel: CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Feb 14 21:34:00 pve kernel: CR2: 00005a9037253508 CR3: 00000001089ca000 CR4: 0000000000f50ef0
Feb 14 21:34:00 pve kernel: PKRU: 55555554

</p>
</details> 

modprobe vfio-pci # Does nothing
modprobe -r amdgpu # Does nothing
```
Basically it gets it to a state where nothing works, I can't attach it back to amdgpu and I have to reboot the host.


Any good place to restart my validation process?

**Comment by BruTTus** [comment]
@drvetcosta would you please help me with a specific step by step guide? I followed all the recommendations but I'm still getting "error 43", this are the status of my W11 VM:

![Image](https://github.com/user-attachments/assets/d3ed401d-d63f-4c3c-92f2-c54665883c5e)

and this is the VM hardware configuration:

![Image](https://github.com/user-attachments/assets/4bee9a57-5823-47ee-98ed-2f0e67cdfa80)

did I miss a BIOS upgrade that I should do to my Minis Forum 890 Pro?


I really appreciate help in this maner, thanks!

**Comment by FrBillyD** [comment]
Hello @BruTTus 👋
Did you find a solution? I have the same problem as you.
